### PR TITLE
Multiple output (nested) grids from fv3atm

### LINKED ISSUE #481: Multiple output grids
## Description
fv3atm should support outputting history files for parent and all nested domains 

## Solution
Update fv3 cap and write grid component to enable more than one set of history files to be written out.

## Alternatives
Use FMS diag manager to write out outputs on all nested domains.

## Testing:

1. Have you tested the code changes? Yes On what platforms? Hera.
2. Have you run regression test in ufs-weather-model or ufs-s2s-model with code changes? Yes/
- Will the baseline results change? Yes.
- If yes, please give brief explanation of your code updates on the regression test. Nested HAFS tests will now have history files written for each domain.

## Dependent PRs:

Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- required to support noaa-emc/GFDL_atmos_cubed_sphere/issues/<issue_number>
- ncar/ccpp-physics/pull/<pr_number>
- associated ufs-weather-model/pull/<pr_number>

### PR DESCRIPTION
**Pull request by DusanJovic-NOAA**
# PR Checklist

- [x] This PR is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR. Please consult the ufs-weather-model [wiki](https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents) if you are unsure how to do this.

- [x] This PR has been tested using a branch which is up-to-date with the top of all sub-component repositories except for those sub-components which are the subject of this PR

- [x] An Issue describing the work contained in this PR has been created either in the subcomponent(s) or in the ufs-weather-model. The Issue should be created in the repository that is most relevant to the changes in contained in the PR. The Issue and the dependent sub-component PR 
are specified below.

- [x] Results for one or more of the regression tests change and the reasons for the changes are understood and explained below.

- [ ] New or updated input data is required by this PR. If checked, please work with the code managers to update input data sets on all platforms.

## Instructions: All subsequent sections of text should be filled in as appropriate.

The information provided below allows the code managers to understand the changes relevant to this PR, whether those changes are in the ufs-weather-model repository or in a subcomponent repository. Ufs-weather-model code managers will use the information provided to add any applicable labels, assign reviewers and place it in the Commit Queue. Once the PR is in the Commit Queue, it is the PR owner's responsiblity to keep the PR up-to-date with the develop branch of ufs-weather-model. 

## Description

This PR updates fv3atm to support writing multiple history files for parent and nested domains. Nested HAFS regression tests were updated to write out history files from all nested domains. 

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues must always be created before starting work on a PR branch!) 
- fixes https://github.com/NOAA-EMC/fv3atm/issues/481

## Testing

How were these changes tested? What compilers / HPCs was it tested with? Are the changes covered by regression tests? (If not, why? Do new tests need to be added?) Have regression tests and unit tests (utests) been run? On which platforms and with which compilers? (Note that unit tests can only be run on tier-1 platforms)

- [x] hera.intel
- [x] hera.gnu
- [x] orion.intel
- [x] cheyenne.intel 
- [x] cheyenne.gnu
- [x] gaea.intel 
- [x] jet.intel
- [x] wcoss_cray
- [x] wcoss_dell_p3
- [ ] opnReqTest for newly added/changed feature
- [x] CI

## Dependencies

If testing this branch requires non-default branches in other repositories, list them. Those branches should have matching names (ideally).

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on https://github.com/NOAA-EMC/fv3atm/pull/480

### CODE REVIEW

File: tests/parm/model_configure_hafs_shared.IN [hunk 1]
```diff
@@ -25,19 +25,126 @@ kchunk3d:                -1
...
-output_grid:             @[OUTPUT_GRID]
-cen_lon:                 @[CEN_LON]             # central longitude
-cen_lat:                 @[CEN_LAT]             # central latitude
-lon1:                    @[LON1]                # longitude of lower-left
-lat1:                    @[LAT1]                # latitude of lower-left
-lon2:                    @[LON2]                # longitude of upper-right
-lat2:                    @[LAT2]                # latitude of upper-right
-dlon:                    @[DLON]
-dlat:                    @[DLAT]
+output_grid:             @[OUTPUT_GRID]     # gaussian_grid, global_latlon, regional_latlon, rotated_latlon, lambert_conformal
```
Comment (BinLiu-NOAA) [inline tests/parm/model_configure_hafs_shared.IN]: @DusanJovic-NOAA, could you please help change this line by adding the **cubed_sphere_grid** in the comment, e.g.,
```
output_grid:             @[OUTPUT_GRID]     # gaussian_grid, global_latlon, regional_latlon, rotated_latlon, lambert_conformal, cubed_sphere_grid
```
Thanks!

Comment (DusanJovic-NOAA) [reply tests/parm/model_configure_hafs_shared.IN]: Done. c327a46

**Review by BinLiu-NOAA — APPROVED** [review]
Changes look good to me. Looking forward to having this capability available in the develop branch.

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: BL
[BL] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/845848280/20220214084506/ufs-weather-model
[BL] Baseline creation and move successful
[RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/845848280/20220214095929/ufs-weather-model
[RT] Error: Test hafs_regional_telescopic_2nests_atm 102 failed in run_test failed 
Please make changes and add the following label back: orion-intel-BL

**Comment by DusanJovic-NOAA** [comment]
> _**Automated RT Failure Notification**_ Machine: orion Compiler: intel Job: BL [BL] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/845848280/20220214084506/ufs-weather-model [BL] Baseline creation and move successful [RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/845848280/20220214095929/ufs-weather-model [RT] Error: Test hafs_regional_telescopic_2nests_atm 102 failed in run_test failed Please make changes and add the following label back: orion-intel-BL

```
$ tail /work/noaa/nems/emc.nemspara/autort/pr/845848280/20220214095929/ufs-weather-model/tests/log_orion.intel/run_102_hafs_regional_telescopic_2nests_atm.log 
+ '[' -o xtrace ']'
+ set_x='set -x'
+ set +x

baseline dir = /work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20220214/INTEL/hafs_regional_telescopic_2nests_atm
working dir  = /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_341631/hafs_regional_telescopic_2nests_atm
Checking test 102 hafs_regional_telescopic_2nests_atm results ....
 Comparing atmf006.nc .........OK
 Comparing sfcf006.nc .........OK
 Comparing atm.nest02.f006.nc ............ALT CHECK......ERROR

```

**Comment by BrianCurtis-NOAA** [comment]
***Automated RT Failure Notification***
Machine: orion
Compiler: intel
Job: RT
[RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/845848280/20220214114530/ufs-weather-model
[RT] Error: Test hafs_regional_telescopic_2nests_atm 102 failed in run_test failed 
Please make changes and add the following label back: orion-intel-RT

**Comment by DusanJovic-NOAA** [comment]
> _**Automated RT Failure Notification**_ Machine: orion Compiler: intel Job: RT [RT] Repo location: /work/noaa/nems/emc.nemspara/autort/pr/845848280/20220214114530/ufs-weather-model [RT] Error: Test hafs_regional_telescopic_2nests_atm 102 failed in run_test failed Please make changes and add the following label back: orion-intel-RT

Same error as in the previous rt run:
```
$ tail /work/noaa/nems/emc.nemspara/autort/pr/845848280/20220214114530/ufs-weather-model/tests/log_orion.intel/run_102_hafs_regional_telescopic_2nests_atm.log 
+ '[' -o xtrace ']'
+ set_x='set -x'
+ set +x

baseline dir = /work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20220214/INTEL/hafs_regional_telescopic_2nests_atm
working dir  = /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_290441/hafs_regional_telescopic_2nests_atm
Checking test 102 hafs_regional_telescopic_2nests_atm results ....
 Comparing atmf006.nc .........OK
 Comparing sfcf006.nc .........OK
 Comparing atm.nest02.f006.nc ............ALT CHECK......ERROR
```
compare_ncfile.py script fails with this error:
```
$ ./tests/compare_ncfile.py /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_290441/hafs_regional_telescopic_2nests_atm/atm.nest02.f006.nc /work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20220214/INTEL/hafs_regional_telescopic_2nests_atm/atm.nest02.f006.nc
Traceback (most recent call last):
  File "./tests/compare_ncfile.py", line 14, in <module>
    if np.shape(nc1[varname][:])!=np.shape(nc2[varname][:]):
  File "netCDF4/_netCDF4.pyx", line 4408, in netCDF4._netCDF4.Variable.__getitem__
  File "netCDF4/_netCDF4.pyx", line 5352, in netCDF4._netCDF4.Variable._get
  File "netCDF4/_netCDF4.pyx", line 1887, in netCDF4._netCDF4._ensure_nc_success
RuntimeError: NetCDF: HDF error
```
h5diff reports no difference:
```
$ h5diff /work/noaa/stmp/bcurtis/stmp/bcurtis/FV3_RT/rt_290441/hafs_regional_telescopic_2nests_atm/atm.nest02.f006.nc /work/noaa/nems/emc.nemspara/RT/NEMSfv3gfs/develop-20220214/INTEL/hafs_regional_telescopic_2nests_atm/atm.nest02.f006.nc
```

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**